ambry-rest/src/main/java/com.github.ambry.rest/AsyncRequestResponseHandler.java

```diff
@@ -61,10 +61,25 @@
   /**
    * Builds a AsyncRequestResponseHandler.
    * @param metrics the {@link RequestResponseHandlerMetrics} instance to use to track metrics.
+   * @param workerCount the required number of request handling units.
+   * @param restRequestService the {@link RestRequestService} instance to be used to process requests.
+   * @throws IllegalArgumentException if {@code workerCount} < 0 or if {@code workerCount} > 0 but
+   * {@code restRequestService} is null.
    */
-  protected AsyncRequestResponseHandler(RequestResponseHandlerMetrics metrics) {
+  protected AsyncRequestResponseHandler(RequestResponseHandlerMetrics metrics, int workerCount,
+      RestRequestService restRequestService) {
     this.metrics = metrics;
     metrics.trackAsyncRequestResponseHandler(this);
+    if (isRunning()) {
```

Comment:
Could this condition ever occur since this is logic is now in the constructor and isRunning is a non-static member?

Reply:
It won't happen. Removing.